lib/StackAnalysis/StackAnalysis.cpp

```diff
@@ -383,6 +682,1178 @@ void commitToModel(GeneratedCodeBasicInfo &GCBI,
+}
+
+template<class FO>
+llvm::Function *CFEPAnalyzer<FO>::outlineFunction(llvm::BasicBlock *Entry) {
+  using namespace llvm;
+  Function *Root = Entry->getParent();
+
+  OnceQueue<BasicBlock *> Queue;
+  std::vector<BasicBlock *> BlocksToClone;
+  Queue.insert(Entry);
+
+  while (!Queue.empty()) {
+    BasicBlock *Current = Queue.pop();
+    BlocksToClone.emplace_back(Current);
+
+    if (isFunctionCall(Current)) {
+      auto *Succ = getFallthrough(Current);
+      auto *Next = getFunctionCallCallee(Current);
+      auto PCCallee = Next ? getBasicBlockPC(Next) : MetaAddress::invalid();
+
+#ifdef NORETURN_IS_FIXED
+      if (Oracle.getFunctionType(PCCallee) != FunctionTypeValue::NoReturn)
+#else
+      if (true)
+#endif
+        Queue.insert(Succ);
+      continue;
+    }
+
+    for (auto *Succ : successors(Current)) {
+      if (!GCBI::isPartOfRootDispatcher(Succ))
+        Queue.insert(Succ);
+    }
+  }
+
+  ValueToValueMapTy VMap;
+  SmallVector<BasicBlock *, 8> BlocksToExtract;
+  AnyPCCloned = UnexpectedPCCloned = nullptr;
+
+  for (const auto &BB : BlocksToClone) {
+    auto *Cloned = CloneBasicBlock(BB, VMap, Twine("__cloned"), Root);
+
+    if (BB == GCBI->anyPC())
+      AnyPCCloned = Cloned;
+    else if (BB == GCBI->unexpectedPC())
+      UnexpectedPCCloned = Cloned;
+
+    VMap[BB] = Cloned;
+    BlocksToExtract.emplace_back(Cloned);
+  }
+
+  remapInstructionsInBlocks(BlocksToExtract, VMap);
+
+  // Fix successor when encountering a call-site and fix fall-through in
+  // presence of a noreturn function.
+  std::map<llvm::CallInst *, MetaAddress> CallMap;
+  for (const auto &BB : BlocksToExtract) {
+    if (isFunctionCall(BB)) {
+      auto *Term = BB->getTerminator();
+      auto *CI = getFunctionCall(Term);
+
+      // If the function callee is null, we are dealing with an indirect call
+      auto *Next = getFunctionCallCallee(Term);
+      auto PCCallee = Next ? getBasicBlockPC(Next) : MetaAddress::invalid();
+      auto CalleeType = Oracle.getFunctionType(PCCallee);
+
+      if (CalleeType != FunctionTypeValue::NoReturn) {
+        auto *Br = BranchInst::Create(getFallthrough(Term));
+        ReplaceInstWithInst(Term, Br);
+      } else if (CalleeType == FunctionTypeValue::NoReturn) {
+#ifdef NORETURN_IS_FIXED
+        auto *Abort = CallInst::Create(M.getFunction("abort"));
+        new UnreachableInst(Term->getContext(), BB);
+        ReplaceInstWithInst(Term, Abort);
+#else
+        auto *Br = BranchInst::Create(getFallthrough(Term));
+        ReplaceInstWithInst(Term, Br);
+#endif
+      }
+
+      // Fix function_call marker
+      unsigned ArgNo = 0;
+      auto *I8PtrTy = Type::getInt8PtrTy(M.getContext());
+      auto *I8NullPtr = ConstantPointerNull::get(I8PtrTy);
+      for (Value *Arg : CI->args()) {
+        if (isa<BlockAddress>(Arg)) {
+          CI->setArgOperand(ArgNo, I8NullPtr);
+          if (Arg->use_empty())
+            cast<BlockAddress>(Arg)->destroyConstant();
+        }
+        ++ArgNo;
+      }
```

Comment:
If you want, you can use `operands` which will return a `Use *` and you can then use `Use::set`.